code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -74,29 +82,44 @@
 		eat_plants()
 
 /mob/living/simple_animal/hostile/retaliate/goat/proc/eat_plants()
-	var/eaten = FALSE
-	var/obj/structure/spacevine/SV = locate(/obj/structure/spacevine) in loc
-	if(SV)
-		SV.eat(src)
-		eaten = TRUE
+	var/obj/structure/spacevine/vine = locate(/obj/structure/spacevine) in loc
+	if(vine)
+		vine.eat(src)
 
-	var/obj/structure/glowshroom/GS = locate(/obj/structure/glowshroom) in loc
-	if(GS)
-		qdel(GS)
-		eaten = TRUE
+	var/obj/structure/alien/resin/flower_bud/flower = locate(/obj/structure/alien/resin/flower_bud) in loc
+	if(flower)
+		flower.take_damage(rand(30, 50), BRUTE, 0)
 
-	if(eaten && prob(10))
-		say("Nom")
+	var/obj/structure/glowshroom/mushroom = locate(/obj/structure/glowshroom) in loc
+	if(mushroom)
+		qdel(mushroom)
+
+	if((vine || flower || mushroom) && prob(10))
+		say("Nom") // bon appetit
+		playsound(src, 'sound/items/eatfood.ogg', rand(30, 50), TRUE)
 
 /mob/living/simple_animal/hostile/retaliate/goat/AttackingTarget()
 	. = ..()
-	if(. && ishuman(target))
-		var/mob/living/carbon/human/H = target
-		if(istype(H.dna.species, /datum/species/pod))
-			var/obj/item/bodypart/NB = pick(H.bodyparts)
-			H.visible_message(span_warning("[src] takes a big chomp out of [H]!"), \
-								  span_userdanger("[src] takes a big chomp out of your [NB]!"))
-			NB.dismember()
+
+	if(!. || !isliving(target))
+		return
+
+	var/mob/living/plant_target = target
+	if(!(plant_target.mob_biotypes & MOB_PLANT))
+		return
+
+	plant_target.adjustBruteLoss(20)
+	playsound(src, 'sound/items/eatfood.ogg', rand(30, 50), TRUE)
+	var/obj/item/bodypart/edible_bodypart
```

Comment:
I deliberately moved this code logic out of the human check so that it affects simple mobs that have the plant biotype.  The human check is only to dismember the limbs.  We still want to deal brute damage, play the sound, and show the message.